portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,52 +42,69 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    UserService userService = UserServiceFactory.getUserService();
+    JSONObject commentsJson = new JSONObject();
+    if (userService.isUserLoggedIn()) {
```

Comment:
An important part of code style/readability is how to structure functions so that they are readable.

Right now you have:

```
if (condition) {
  // 35
  // lines
  // ...
} else {
  // some other code
}
```

Instead, you should create a helper function - say, `getLoggedInComments()`, so you can collapse this into:

```
if (condition) {
  commentsJson = getLoggedInComments();
} else {
  commentsJson = getLoggedOutComments();
}
```

This makes the function much easier to read, and also easier to scan without needing to scroll up a bunch to check the conditions and indent level.